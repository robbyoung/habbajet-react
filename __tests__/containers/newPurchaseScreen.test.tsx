import React from 'react';
import renderer from 'react-test-renderer';
import store from '../../app/store';
import {Provider} from 'react-redux';
import NewPurchaseScreen from '../../app/containers/newPurchaseScreen';

jest.mock('@fortawesome/react-native-fontawesome', () => ({
    FontAwesomeIcon: '',
}));

jest.mock('../../app/storage', () => ({
    saveState: () => undefined,
}));

describe('New Purchase Screen Component', () => {
    it('will map state to a purchase creation form', () => {
        const component = renderer.create(
            <Provider store={store}>
                <NewPurchaseScreen />
            </Provider>,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});
