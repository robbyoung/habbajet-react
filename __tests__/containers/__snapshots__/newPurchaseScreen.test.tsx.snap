// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Purchase Screen Component will map state to a purchase creation form 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#ffffff",
      "height": "100%",
      "paddingHorizontal": 20,
      "paddingVertical": 10,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "marginBottom": 15,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#959595",
              "fontFamily": "Abel",
              "fontSize": 20,
              "margin": 0,
              "padding": 0,
            },
          ]
        }
      >
        Name
      </Text>
      <TextInput
        allowFontScaling={true}
        keyboardType="default"
        onChangeText={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#959595",
            "borderWidth": 1,
            "fontFamily": "Abel",
            "fontSize": 30,
            "paddingLeft": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View />
    </View>
    <View
      style={
        Object {
          "marginBottom": 15,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#959595",
              "fontFamily": "Abel",
              "fontSize": 20,
              "margin": 0,
              "padding": 0,
            },
          ]
        }
      >
        Cost
      </Text>
      <TextInput
        allowFontScaling={true}
        keyboardType="numeric"
        onChangeText={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#959595",
            "borderWidth": 1,
            "fontFamily": "Abel",
            "fontSize": 30,
            "paddingLeft": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#959595",
          "borderRadius": 10,
          "marginBottom": 10,
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
              "color": "#ffffff",
              "fontFamily": "Abel",
              "fontSize": 30,
              "padding": 10,
              "textAlign": "center",
              "width": "100%",
            },
          ]
        }
      >
        Done
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
